exemplar/extractor.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package exemplar implements support for exemplars. Exemplars are additional
+// data associated with each measurement.
+//
+// Their purpose it to provide an example of the kind of thing
+// (request, RPC, trace span, etc.) that resulted in that measurement.
+package exemplar
+
+import (
+	"context"
+	"time"
+)
+
+// Exemplar represents an abstract set of supplemental information associated with
+// a measurement, for the purpose of providing an example of the type of
+// situation that resulted in the measurement.
+// For example, a trace exemplar contains key-value pairs representing a
+// SpanContext and points to an example span that resulted in a given
+// measurement.
+type Exemplar struct {
```

Comment:
Am curious why we have two very similar structs `Exemplar` and `ExemplarData` which vary by only one field `Value`, but also this definition of Exemplar seems only suited to traces yet we expect them to work for both traces and metrics.

Reply:
Done